Dependency Conflict Error During Vercel Deployment (Works Locally)

<!--
Please read and follow the issue template.
Issues submitted without a reproduction and context
will take longer to resolve.
-->

## Context

What's your version of `nuqs`?

```
-> "nuqs": "^2.1.1"
```

What framework are you using?

- ✅ Next.js (app router)
- ❌ Next.js (pages router)
- ❌ React SPA (no router)
- ❌ Remix
- ❌ React Router
- ❌ Other (please specify)

Which version of your framework are you using?

```
-> "next": "15.0.3"
-> "react": "19.0.0-rc-66855b96-20241106"
-> "react-dom": "19.0.0-rc-66855b96-20241106"
```

## Description

My project is encountering a dependency conflict error during deployment on Vercel, specifically related to React versions. The project works fine locally, but fails during the Vercel build process. The error suggests a conflict between the pre-release version of React 19 we're using and the requirements of other packages, particularly `nuqs` and `@radix-ui` components.

## Reproduction

Steps to reproduce the behavior:

1. Push changes to the connected GitHub repository
2. Trigger a deployment on Vercel (automatically or manually)
3. Observe the build fail with the following error:

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: nuqs@2.1.1
npm error Found: react@19.0.0-rc-66855b96-20241106
npm error node_modules/react
npm error   react@"19.0.0-rc-66855b96-20241106" from the root project
npm error   peer react@"^16.8 || ^17.0 || ^18.0 || ^19.0 || ^19.0.0-rc" from @radix-ui/react-arrow@1.1.0
npm error   node_modules/@radix-ui/react-arrow
npm error     @radix-ui/react-arrow@"1.1.0" from @radix-ui/react-popper@1.2.0
npm error     node_modules/@radix-ui/react-popper
npm error       @radix-ui/react-popper@"1.2.0" from @radix-ui/react-menu@2.1.2
npm error       node_modules/@radix-ui/react-menu
npm error         @radix-ui/react-menu@"2.1.2" from @radix-ui/react-dropdown-menu@2.1.2
npm error         node_modules/@radix-ui/react-dropdown-menu
npm error   38 more (@radix-ui/react-avatar, @radix-ui/react-collection, ...)
npm error
npm error Could not resolve dependency:
npm error peer react@">= 18.2.0" from nuqs@2.1.1
npm error node_modules/nuqs
npm error   nuqs@"^2.1.1" from the root project
npm error
npm error Conflicting peer dependency: react@18.3.1
npm error node_modules/react
npm error   peer react@">= 18.2.0" from nuqs@2.1.1
npm error   node_modules/nuqs
npm error     nuqs@"^2.1.1" from the root project
```

Additional context:
- The project builds and runs without issues in the local development environment.
- We're using a pre-release version of React 19, which seems to be the root cause of the conflict.

Any insights or suggestions on resolving this deployment-specific dependency conflict would be greatly appreciated. Thank you!

--- a/packages/nuqs/package.json
+++ b/packages/nuqs/package.json
@@ -114,10 +114,10 @@
     "mitt": "^3.0.1"
   },
   "peerDependencies": {
-    "@remix-run/react": ">= 2",
-    "next": ">= 14.2.0",
-    "react": ">= 18.2.0",
-    "react-router-dom": ">= 6"
+    "@remix-run/react": ">=2",
+    "next": ">=14.2.0",
+    "react": ">=18.2.0",
+    "react-router-dom": ">=6"
   },
   "peerDependenciesMeta": {
     "@remix-run/react": {

--- a/packages/nuqs/package.json
+++ b/packages/nuqs/package.json
@@ -116,7 +116,7 @@
   "peerDependencies": {
     "@remix-run/react": ">=2",
     "next": ">=14.2.0",
-    "react": ">=18.2.0",
+    "react": ">=18.2.0 || ^19.0.0-0",
     "react-router-dom": ">=6"
   },
   "peerDependenciesMeta": {

